azure-core/README.md

```diff
@@ -0,0 +1,357 @@
+
+    def __init__(self, **kwargs):
+        self.timeout = kwargs.pop('connection_timeout', 100)
+        self.verify = kwargs.pop('connection_verify', True)
+        self.cert = kwargs.pop('connection_cert', None)
+        self.data_block_size = kwargs.pop('connection_data_block_size', 4096)
+        self.keep_alive = kwargs.pop('connection_keep_alive', False)
+```
+Currently we are also using transport configuration for Proxy support - although this may at some point be turned into a dedicated policy.
+
+### HttpRequest and HttpResponse
+
+The HttpRequest and HttpResponse objects represent a generic concept of HTTP request and response constructs and are in no way tied to a particular transport or HTTP library.
+
+The HttpRequest has the following API. It does not vary between transports:
+```python
+class HttpRequest(object):
+    
+    def __init__(self, method, url, headers=None, files=None, data=None):
+        self.method = method
+        self.url = url
+        self.headers = CaseInsensitiveDict(headers)
+        self.files = files
+        self.data = data
+
+    @property
+    def body(self):
+        return self.data
+
+    @body.setter
+    def body(self, value):
+        self.data = value
+
+    def format_parameters(self, params):
+        """Format parameters into a valid query string.
+        It's assumed all parameters have already been quoted as
+        valid URL strings."""
+
+    def add_content(self, data):
+        """Add a body to the request."""
+
+    def add_formdata(self, content=None):
+        """Add data as a multipart form-data request to the request."""
+```
+
+The HttpResponse object on the other hand will generally have a transport-specific derivative.
+This is to accomodate how the data is extracted for the object returned by the HTTP library.
+There is also an async flavor: AsyncHttpResponse. This is to allow for the asynchronous streaming of
+data from the response.
+For example:
+```python
+from azure.core.pipeline.transport import (
+    RequestsTransportResponse,  # HttpResponse
+    AioHttpTransportResponse, # AsyncHttpResponse
+    TrioRequestsTransportResponse,  # AsyncHttpResponse
+    AsyncioRequestsTransportResponse,  # AsyncHttpResponse
+)
+```
+The API for each of these response types is identical, so the consumer of the Response need not know about these
+particular types.
+
+The HttpResponse has the following API. It does not vary between transports:
+```python
+class HttpResponse(object):
+
+    def __init__(self, request, internal_response):
+        self.request = request
+        self.internal_response = internal_response  # The object returned be the HTTP library
+        self.status_code = None
+        self.headers = {}
+        self.reason = None 
+
+    def body(self):
+        """Return the whole body as bytes in memory."""
+
+    def text(self, encoding=None):
+        """Return the whole body as a string."""
+
+    def raise_for_status(self):
```

Comment:
Do we need this method?

Reply:
Probably not.... is likely there for previous msrest implementation to allow the object to behave like a Requests response object. I don't think it's being used anywhere.